aws/lambda/oss_ci_job_queue_time/lambda_function.py

```diff
@@ -0,0 +1,699 @@
+        else:
+            runner_labels_dict["nonephemeral"].add(runner)
+
+        if runner_config["os"].lower() == "linux":
+            runner_labels_dict["linux"].add(runner)
+        elif runner_config["os"].lower() == "windows":
+            runner_labels_dict["windows"].add(runner)
+
+    for runner, runner_config in runner_configs["runner_types"].items():
+        runner_labels_dict["meta"].add(runner)
+
+        if runner_config["os"].lower() == "linux":
+            runner_labels_dict["linux-meta"].add(runner)
+        elif runner_config["os"].lower() == "windows":
+            runner_labels_dict["windows-meta"].add(runner)
+
+    for runner, runner_config in lf_runner_configs["runner_types"].items():
+        runner_labels_dict["lf"].add(runner)
+
+        if runner_config["os"].lower() == "linux":
+            runner_labels_dict["linux-lf"].add(runner)
+        elif runner_config["os"].lower() == "windows":
+            runner_labels_dict["windows-lf"].add(runner)
+    return runner_labels_dict
+
+
+def get_runner_config(
+    retriever: LazyFileHistory, start_time: str | datetime
+) -> Dict[str, Dict[str, Any]]:
+    contents = retriever.get_version_after_timestamp(start_time)
+    if contents:
+        return explode_runner_variants(yaml.safe_load(contents))
+    return {"runner_types": {}}
+
+
+def get_config_retrievers(github_access_token: str) -> Tuple[Any, Any, Any]:
+    auth = Auth.Token(github_access_token)
+    test_infra_repo = Github(auth=auth).get_repo("pytorch/test-infra")
+    pytorch_repo = Github(auth=auth).get_repo("pytorch/pytorch")
+
+    meta_runner_config_retriever = LazyFileHistory(
+        test_infra_repo, ".github/scale-config.yml"
+    )
+    lf_runner_config_retriever = LazyFileHistory(
+        test_infra_repo, ".github/lf-scale-config.yml"
+    )
+    old_lf_lf_runner_config_retriever = LazyFileHistory(
+        pytorch_repo, ".github/lf-scale-config.yml"
+    )
+
+    return (
+        meta_runner_config_retriever,
+        lf_runner_config_retriever,
+        old_lf_lf_runner_config_retriever,
+    )
+
+
+class QueueTimeProcessor:
+    """
+    this class used to handle oss ci queue time data aggregations. Currently it fetches in-queue jobs from clickhouse at current time
+
+    To run the main method:
+       processor = QueueTimeProcessor(clickhouse_client,s3_client)
+       processor.process()
+    """
+
+    def __init__(
+        self,
+        clickhouse_client: Any,
+        s3_client: Any,
+        is_dry_run: bool = False,
+        local_output: bool = False,
+    ) -> None:
+        self.clickhouse_client = clickhouse_client
+        self.s3_client = s3_client
+        self.is_dry_run = is_dry_run
+        self.local_output = local_output and is_dry_run
+
+    def process(self) -> None:
+        github_access_token = os.getenv("GITHUB_ACCESS_TOKEN", "")
+        if not github_access_token:
+            raise ValueError("Missing environment variable GITHUB_ACCESS_TOKEN")
```

Comment:
This should be a parameter for the class, and the validation should fail on init